Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/PaintBrushManipulator.cpp

```diff
@@ -137,12 +151,29 @@ namespace AzToolsFramework
         {
             if (mouseInteraction.m_mouseInteraction.m_mouseButtons.Left())
             {
-                m_isPainting = false;
-                PaintBrushNotificationBus::Event(m_ownerEntityComponentId, &PaintBrushNotificationBus::Events::OnPaintEnd);
+                // Notify that the paint stroke has ended.
+                // We need to verify that we're currently painting, because clicks/double-clicks can cause us to end up here
+                // without ever getting the mouse Down event.
+                if (m_isPainting)
+                {
+                    m_isPainting = false;
+                    PaintBrushNotificationBus::Event(m_ownerEntityComponentId, &PaintBrushNotificationBus::Events::OnPaintStrokeEnd);
+                }
 
                 return true;
             }
         }
+        else if (mouseInteraction.m_mouseEvent == AzToolsFramework::ViewportInteraction::MouseEvent::DoubleClick)
+        {
+            if (mouseInteraction.m_mouseInteraction.m_mouseButtons.Left())
+            {
+                // Swallow up double-click events so that the default component mode behavior doesn't detect it and try
+                // to leave component mode. It's too easy to double-click by accident when painting, so double-clicking
+                // to exit just doesn't work well.
+                return true;
+            }
+        }
```

Comment:
question: If this is the case maybe we should do away with this behavior entirely now otherwise being inconsistent across Component Modes could be confusing. Maybe something to ask @o3de/sig-ui-ux about?

Reply:
I'll be discussing all of the recent paintbrush changes with UX later this afternoon, I'll make sure to ask about this as well.

Reply:
I asked about this in the sig-ui-ux office hours just now, and it sounds like there's _already_ been conversations about removing the default double-click-to-exit behavior. So as long as you're good with it, I'll put up a separate PR to remove that default behavior, along with my special-case override here, since it will no longer be necessary.

Reply:
Sounds good to me thanks!